pinot-query-runtime/src/test/resources/queries/LexicalStructure.json

```diff
@@ -0,0 +1,184 @@
+        "schema": [{"name": "U&\"d\\0061t\\+000061\"", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [
+      {
+        "psql": "4.1.1",
+        "description": "quoted identifier support",
+        "sql": "SELECT * FROM {tbl}"
+      }
+    ]
+  },
+  "c_string_constants": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": ["4.1.2.1", "4.1.2.2"],
+      "description": "constants with c-style escapes",
+      "sql": "SELECT data, 'foo', 'foo\nbar', concat(data, 'foo\tbar', '') FROM {tbl}"
+    }]
+  },
+  "unicode_string_constants": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.2.3",
+      "description": "constants with unicode escapes",
+      "sql": "SELECT data, U&'d\\0061ta' FROM {tbl}",
+      "outputs": [["1", "data"]]
+    }]
+  },
+  "dollar_string_constants": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.2.4",
+      "ignored": true,
+      "comment": "unsupported",
+      "description": "constants with $ escapes",
+      "sql": "SELECT data, $$Dianne's horse$$, $tag$Dianne's\nhorse$tag$ FROM {tbl}"
+    }]
+  },
+  "bit_string_constants": {
+    "ignored": true,
+    "comment": "bit string constants not supported",
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.2.5",
+      "description": "bit string constant support",
+      "sql": "SELECT data, B'1001', X'1FF' FROM {tbl}",
+      "outputs": [["1", "bytes", "bytes"]]
+    }]
+  },
+  "numeric_constants": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.2.6",
+      "description": "numeric constant supports",
+      "sql": "SELECT data, 42, 3.5, 4., .001, 5e2, 1.925e-3 FROM {tbl}"
+    }]
+  },
+  "constant_casting": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.2.7",
+      "description": "constant casting",
+      "sql": "SELECT data, CAST ('42' AS INT) FROM {tbl}"
+    }]
+  },
+  "comments": {
+    "tables": {
+      "tbl": {
+        "schema": [{"name": "data", "type": "STRING"}],
+        "inputs": [["1"]]
+      }
+    },
+    "queries": [{
+      "psql": "4.1.5",
+      "description": "testing comment support",
+      "sql": "---this is a comment\nSELECT data FROM {tbl}"
+    }]
+  },
+  "operator_precedence" : {
+    "comments": "we don't support ^ or array[x] yet in v2, unary minus is broken",
```

Comment:
this comment seems to indicate we should ignore? but not sure if this is correct

Reply:
it's a partial test - I didn't add those operations in so I couldn't cover all operator precedence but I could cover a lot

Reply:
got it. so e.g. the ones should be ignored are not listed in the queries list below.